src/client/Microsoft.Identity.Client/ManagedIdentity/AppServiceManagedIdentitySource.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Client.ManagedIdentity
+{
+    internal class AppServiceManagedIdentitySource : ManagedIdentitySource
+    {
+        // MSI Constants. Docs for MSI are available here https://docs.microsoft.com/azure/app-service/overview-managed-identity
+        protected const string AppServiceMsiApiVersion = "2019-08-01";
+        protected const string SecretHeaderName = "X-IDENTITY-HEADER";
```

Comment:
There is a lot of use of `protected`. Does this help with testing?

Reply:
I changed it to private. It was protected because the original code had implementation for 2 app service. We are not supporting the legacy one.